### PR TITLE
chore(api-client): add more OpenAPI features to workspaces 

### PR DESCRIPTION
**Pull request by geoffgscott**
Preliminary work to align our workspace better with Open API Specification types to allow better bidirectional data translation.

### CODE REVIEW

**Review by marclave — COMMENTED** [review]
could we chunk this PR into a few smaller PRs, seems like the store refactor could be broken up into a refactor outside of this PR. 

hard to see the diffs between old and new store w mix of moving :)

File: .vscode/settings.json [hunk 1]
```diff
@@ -53,5 +53,11 @@
   },
   "typescript.tsdk": "node_modules/typescript/lib",
   "vitest.commandLine": "pnpm vitest",
-  "vitest.enable": true
+  "vitest.enable": true,
+  "yaml.schemas": {
+    "https://raw.githubusercontent.com/OAI/OpenAPI-Specification/main/schemas/v3.1/schema.json": [
+      "file:///home/geoff/gitpro/scalar/packages/oas-utils/src/diff/fixtures/a-basic.test.yaml",
+      "file:///home/geoff/gitpro/scalar/packages/oas-utils/src/diff/fixtures/b-basic-add.test.yaml"
```
Comment (hanspagel) [inline .vscode/settings.json:60]: This has some paths to @geoffgscott’s machine 🤓

File: packages/api-client/src/components/Sidebar/Actions/SidebarListElementForm.vue [hunk 2]
```diff
@@ -12,7 +12,7 @@ const emit = defineEmits<{
 }>()
 </script>
 <template>
-  <form
+  <div
```
[the remark this answers is not in the document]
Comment (geoffgscott) [reply packages/api-client/src/components/Sidebar/Actions/SidebarListElementForm.vue:15]: Using form without using the actual form object is anti-pattern IMO

[the remark this answers is not in the document]
Comment (hanspagel) [reply packages/api-client/src/components/Sidebar/Actions/SidebarListElementForm.vue:15]: Interesting! I see it the other way around. Not using a form makes it harder for password managerd and screen readers, no matter if we’re sending the form the regular way or not.

AFAIK the form is - semantically - just grouping inputs, so the browser, password managers, screen readers … can make sense of it.

**Comment by hanspagel** [comment]
- [ ] rebasing on main
- [ ] restore File type changes in packages/oas-utils/src/entities/spec/request-examples.ts
- [ ] merge #3067 to restore changes to sendRequest
- [ ] compare resulting sendRequest tests to main

**Review by marclave — APPROVED** [review]
wagmi

**Pull request merged**